t_sparse_list.py

```diff
@@ -109,6 +109,17 @@ def test_slice_reversal_empty(self):
         sl = sparse_list.SparseList(4)
         self.assertEquals([None, None, None, None], sl[::-1])
 
+    def test_slice_with_list_read(self):
+        sl = sparse_list.SparseList([1, 2, 3, 4, 5])
+        ip = [0, 2, 4]
+        self.assertEquals([1, 3, 5], sl[ip])
```

Comment:
Now that you mention it, it would be best for this lib to keep a signature similar to the vanilla list.
This whole PR is useless, because if you want a `numpy`-like signature, there's `sparse_vector` for that now.

I kinda did it along the way, without thinking much.